File: backend-python/ai-service/app/models/model_config.py
"""
模型配置数据库模型
"""
from sqlalchemy import Column, BigInteger, String, Integer, Numeric, DateTime
from sqlalchemy.sql import func
from app.database import Base


class ModelConfig(Base):
    """模型配置表"""
    __tablename__ = "model_config"
    
    id = Column(BigInteger, primary_key=True, index=True)
    model_code = Column(String(100), unique=True, nullable=False, index=True)
    model_name = Column(String(200), nullable=False)
    model_type = Column(String(50))
    api_endpoint = Column(String(500))
    api_key = Column(String(500))
    model_version = Column(String(50))
    max_tokens = Column(Integer)
    temperature = Column(Numeric(3, 2))
    is_active = Column(String(1), default="1")
    priority = Column(Integer)
    daily_limit = Column(Integer)
    create_time = Column(DateTime(timezone=True), server_default=func.now())
    update_time = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
    
    def to_dict(self):
        """转换为字典"""
        return {
            "id": self.id,
            "model_code": self.model_code,
            "model_name": self.model_name,
            "model_type": self.model_type,
            "api_endpoint": self.api_endpoint,
            "api_key": self.api_key,  # 注意：实际使用时可能需要脱敏
            "model_version": self.model_version,
            "max_tokens": self.max_tokens,
            "temperature": float(self.temperature) if self.temperature else None,
            "is_active": self.is_active,
            "priority": self.priority,
            "daily_limit": self.daily_limit,
            "create_time": self.create_time.isoformat() if self.create_time else None,
            "update_time": self.update_time.isoformat() if self.update_time else None,
        }

File: backend-java/test-design-assistant-core/src/main/java/com/sinosoft/testdesign/entity/ModelConfig.java
package com.sinosoft.testdesign.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 模型配置实体
 * 
 * @author sinosoft
 * @date 2024-01-01
 */
@Data
@Entity
@Table(name = "model_config")
public class ModelConfig {
    
    /**
     * 主键ID
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    /**
     * 模型编码
     */
    @Column(name = "model_code", unique = true, nullable = false, length = 100)
    private String modelCode;
    
    /**
     * 模型名称
     */
    @Column(name = "model_name", nullable = false, length = 200)
    private String modelName;
    
    /**
     * 模型类型：DeepSeek/豆包/Kimi/千问/智谱
     */
    @Column(name = "model_type", length = 50)
    private String modelType;
    
    /**
     * API端点
     */
    @Column(name = "api_endpoint", length = 500)
    private String apiEndpoint;
    
    /**
     * API密钥（加密存储）
     */
    @Column(name = "api_key", length = 500)
    private String apiKey;
    
    /**
     * 模型版本
     */
    @Column(name = "model_version", length = 50)
    private String modelVersion;
    
    /**
     * 最大Token数
     */
    @Column(name = "max_tokens")
    private Integer maxTokens;
    
    /**
     * 温度参数
     */
    @Column(name = "temperature", precision = 3, scale = 2)
    private BigDecimal temperature;
    
    /**
     * 是否启用：1-启用，0-禁用
     */
    @Column(name = "is_active", columnDefinition = "CHAR(1)")
    private String isActive = "1";
    
    /**
     * 优先级（数字越小优先级越高）
     */
    @Column(name = "priority")
    private Integer priority;
    
    /**
     * 每日调用限制
     */
    @Column(name = "daily_limit")
    private Integer dailyLimit;
    
    /**
     * 创建时间
     */
    @Column(name = "create_time")
    private LocalDateTime createTime;
    
    /**
     * 更新时间
     */
    @Column(name = "update_time")
    private LocalDateTime updateTime;
    
    @PrePersist
    public void prePersist() {
        this.createTime = LocalDateTime.now();
        this.updateTime = LocalDateTime.now();
    }
    
    @PreUpdate
    public void preUpdate() {
        this.updateTime = LocalDateTime.now();
    }
}

